app/include/dt-bindings/zmk/keys.h

```diff
@@ -6,143 +6,1387 @@
+
+/* Keyboard Return (Enter) */
+#define RETURN (HID_USAGE_KEY_KEYBOARD_RETURN_ENTER)
+#define ENTER (RETURN)
+#define RET (RETURN)
+
+/* Keyboard Escape */
+#define ESCAPE (HID_USAGE_KEY_KEYBOARD_ESCAPE)
+#define ESC (ESCAPE)
+
+/* Keyboard Backspace */
+#define BACKSPACE (HID_USAGE_KEY_KEYBOARD_DELETE_BACKSPACE)
+#define BSPC (BACKSPACE)
+#define BKSP (BACKSPACE) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard Tab */
+#define TAB (HID_USAGE_KEY_KEYBOARD_TAB)
+
+/* Keyboard Space */
+#define SPACE (HID_USAGE_KEY_KEYBOARD_SPACEBAR)
+#define SPC (SPACE) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard - and _ (Minus and Underscore) */
+#define MINUS (HID_USAGE_KEY_KEYBOARD_MINUS_AND_UNDERSCORE)
+
+/* Keyboard _ (Underscore) */
+#define UNDERSCORE (HID_USAGE_KEY_KEYBOARD_MINUS_AND_UNDERSCORE)
+#define UNDER (UNDERSCORE)
+
+/* Keyboard = and + (Equal and Plus) */
+#define EQUAL (HID_USAGE_KEY_KEYBOARD_EQUAL_AND_PLUS)
+#define EQL (EQUAL) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard + (Plus) */
+#define PLUS (HID_USAGE_KEY_KEYBOARD_EQUAL_AND_PLUS)
+
+/* Keyboard [ and { (Left Bracket and Left Brace) */
+#define LEFT_BRACKET (HID_USAGE_KEY_KEYBOARD_LEFT_BRACKET_AND_LEFT_BRACE)
+#define LBKT (LEFT_BRACKET)
+
+/* Keyboard { (Left Brace) */
+#define LEFT_BRACE (HID_USAGE_KEY_KEYBOARD_LEFT_BRACKET_AND_LEFT_BRACE)
+#define LBRC (LEFT_BRACE)
+#define LCUR (LEFT_BRACE) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard ] and } (Right Bracket and Right Brace) */
+#define RIGHT_BRACKET (HID_USAGE_KEY_KEYBOARD_RIGHT_BRACKET_AND_RIGHT_BRACE)
+#define RBKT (RIGHT_BRACKET)
+
+/* Keyboard } (Right Brace) */
+#define RIGHT_BRACE (HID_USAGE_KEY_KEYBOARD_RIGHT_BRACKET_AND_RIGHT_BRACE)
+#define RBRC (RIGHT_BRACE)
+#define RCUR (RIGHT_BRACE) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard \ and | (Backslash and Pipe) */
+#define BACKSLASH (HID_USAGE_KEY_KEYBOARD_BACKSLASH_AND_PIPE)
+#define BSLH (BACKSLASH)
+
+/* Keyboard | (Pipe) */
+#define PIPE (HID_USAGE_KEY_KEYBOARD_BACKSLASH_AND_PIPE)
+
+/* Keyboard Non-US # and ~ (Non-US Hash/Number and Tilde) */
+#define NON_US_HASH (HID_USAGE_KEY_KEYBOARD_NON_US_HASH_AND_TILDE)
+
+/* Keyboard ~ (Tilde) */
+#define TILDE2 (HID_USAGE_KEY_KEYBOARD_NON_US_HASH_AND_TILDE)
+
+/* Keyboard ; and : (Semicolon and Colon) */
+#define SEMICOLON (HID_USAGE_KEY_KEYBOARD_SEMICOLON_AND_COLON)
+#define SEMI (SEMICOLON)
+#define SCLN (SEMICOLON) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard : (Colon) */
+#define COLON (HID_USAGE_KEY_KEYBOARD_SEMICOLON_AND_COLON)
+#define COLN (COLON) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard ' and " (Apostrophe and Quote) */
+#define SINGLE_QUOTE (HID_USAGE_KEY_KEYBOARD_APOSTROPHE_AND_QUOTE)
+#define SQT (SINGLE_QUOTE)
+#define APOSTROPHE (SINGLE_QUOTE)
+#define APOS (SINGLE_QUOTE)
+#define QUOT (SINGLE_QUOTE) // WARNING: DEPRECATED (DO NOT USE)
+
+/* Keyboard " (Quote) */
+#define DOUBLE_QUOTES (HID_USAGE_KEY_KEYBOARD_APOSTROPHE_AND_QUOTE)
```

Comment:
With all of these added "shifted codes", would the replacing these with the shifted varients once that's merged. e.g.:

```suggestion
#define DOUBLE_QUOTES (LS(HID_USAGE_KEY_KEYBOARD_APOSTROPHE_AND_QUOTE))
```

?

That's really my only outstanding item I would want clarity on. Otherwise this looks great.

Reply:
The rebased branch with modifiers and this PR has these keycodes with the shifted variant: https://github.com/innovaker/zmk/tree/hid-keys-modifiers-rebase. I.e. 

https://github.com/innovaker/zmk/blob/6e52af9870ddb88e05a948e4d9160911e6543e9f/app/include/dt-bindings/zmk/keys.h#L278-L280

So I expect that is the plan, though it was emphasised that branch is a "dry run" so there may be some other tiny bits of work around integrating them I guess.

(Caveat aside, I've been using that  branch (hid-keys-modifiers-rebase) all of today with no issues except a few expected collisions between names and my own defines when doing the build, which is expected. Made my keymap a lot shorter, I only need defines for UK keycodes now, as the names don't make much sense there).

Reply:
@CrossR Indeed. Good catch. Thanks.